doc/guides/coding_guidelines/index.rst

```diff
@@ -0,0 +1,778 @@
+.. _coding_guidelines_misra:
+
+Coding Guidelines
+#################
+
+The project TSC and the Safety Committee of the project to implement a staged
+and incremental approach for complying with a set of coding rules (AKA Coding
+Guidelines) to improve quality and consistency of the code base. Below are the
+agreed upon stages and the approximate timelines:
+
+Stage I
+  Coding guideline rules are available to be followed and referenced,
+  but not enforced. Rules are not yet enforced in CI and pull-requests cannot be
+  blocked by reviewers/approvers due to violations.
+
+Stage II
+  Begin enforcement on a limited scope of the code base. Initially this would be
+  safety certification scope. For rules easily applied across codebase, we
+  should not limit compliance to initial scope. This step requires tooling and
+  CI setup.
+  The enforcement for safety certification scope to begin during the 2.4
```

Comment:
s/The enforcement for safety certification scope to begin/This stage will begin during/